benchmark/torch/ddpg/train.py

```diff
@@ -0,0 +1,158 @@
+    parser.add_argument(
+        "--batch_size", default=100, type=int)  # Batch size for learning
+    parser.add_argument("--discount", default=0.99)  # Discount factor
+    parser.add_argument("--tau", default=0.005)  # Target network update rate
+    parser.add_argument(
+        "--actor_lr", default=1e-3)  # Learning rate of actor network
+    parser.add_argument(
+        "--critic_lr", default=1e-3)  # Learning rate of critic network
+    parser.add_argument(
+        "--policy_freq", default=1,
+        type=int)  # Frequency to train actor and update params
+    args = parser.parse_args()
+
+    logger.info("------------------ DDPG ---------------------")
+    logger.info('Env: {}, Seed: {}'.format(args.env, args.seed))
+    logger.info("---------------------------------------------")
+
+    env = gym.make(args.env)
+
+    # Set seeds
+    env.seed(args.seed)
+
+    state_dim = env.observation_space.shape[0]
+    action_dim = env.action_space.shape[0]
+    max_action = float(env.action_space.high[0])
+
+    # Initialize model, algorithm, agent, replay_memory
+    model = MujocoModel(state_dim, action_dim, max_action)
+    algorithm = DDPG(
+        model,
+        max_action,
+        discount=args.discount,
+        tau=args.tau,
+        actor_lr=args.actor_lr,
+        critic_lr=args.critic_lr,
+        policy_freq=args.policy_freq)
+    agent = MujocoAgent(algorithm, state_dim, action_dim)
+    replay_memory = ReplayMemory(
+        max_size=int(1e6), obs_dim=state_dim, act_dim=action_dim)
+
+    obs, done = env.reset(), False
+    episode_reward = 0
+    episode_timesteps = 0
+    episode_num = 0
+
+    for t in range(int(args.max_timesteps)):
+
+        episode_timesteps += 1
+
+        # Select action randomly or according to policy
+        if t < args.start_timesteps:
+            action = env.action_space.sample()
+        else:
+            action = (agent.predict(np.array(obs)) + np.random.normal(
+                0, max_action * args.expl_noise, size=action_dim)).clip(
+                    -max_action, max_action)
+
+        # Perform action
+        next_obs, reward, done, _ = env.step(action)
+        terminal = float(
+            done) if episode_timesteps < env._max_episode_steps else 0
+        terminal = 1. - terminal
+
+        # Store data in replay memory
+        replay_memory.append(obs, action, reward, next_obs, terminal)
+
+        obs = next_obs
+        episode_reward += reward
+
+        # Train agent after collecting sufficient data
+        if t >= args.start_timesteps:
+            batch_obs, batch_action, batch_reward, batch_next_obs, batch_terminal = replay_memory.sample_batch(
+                args.batch_size)
+            agent.learn(batch_obs, batch_action, batch_reward, batch_next_obs,
+                        batch_terminal)
+
+        if done:
+            # +1 to account for 0 indexing. +0 on ep_timesteps since it will increment +1 even if done=True
```

Comment:
You can remove this comment since the log print is clear enough

Reply:
![image](https://user-images.githubusercontent.com/76139596/102599484-7fae7800-4158-11eb-8d08-e5cf88c573c7.png)
Resolved.